do-catalog/src/store/actions/catalog.js

```diff
@@ -127,3 +128,14 @@ export function deleteFilter(context, filter) {
 export function clearTagFilters(context) {
   context.commit('resetTagFilters');
 }
+
+export async function fetchSubscriptionInfo(context, { id, type, api_key }) {
+  let url = `${baseUrl}/${subscriptionsEndpoint}?id=${id}&type=${type}&api_key=${api_key}`;
```

Comment:
Not exactly a frontend issue, but I'm not sure that I'm understanding well how the subscription API is defined.
IMO, the logical way to access to the catalog is through `public.carto.com/...` and make the user related calls (like subscribe, check current subscriptions, etc) through `user.carto.com/...`, so we can secure this endpoints the same way as the rest and apply cache and other scalability optimizations to the catalog that would be the same for all users.
In this case, it looks like both will work from public setting an API key (master?) without using `_cartodb_session` that at first is different to the rest of the endpoints for CARTO.
I may be misunderstanding how the API works or maybe it is a planned change, but I feel that I should point it to be aligned to the rest of the project.

Reply:
After talking with @antoniocarlon the API works split into the user side and the public side, so, we just need to add the "get user base URL" method to the front end.